packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -174,16 +175,23 @@ class TimeInput extends React.Component {
     };
   }
 
-  componentDidUpdate(prevProps) {
-    if (
-      this.props.value === prevProps.value
-      && this.props.variant === prevProps.variant
+  componentDidUpdate(prevProps, prevState) {
+    let skipStateUpdate = true;
+    const prevStateTimeValue = (this.props.showSeconds) ? `${prevState.hour}:${prevState.minute}:${prevState.second}` : `${prevState.hour}:${prevState.minute}`;
+    // Comparing partial time values on each update will result in incorrect comparisons of prevState and propsValue.
+    // Hence validating propsValue and prevState values to skip the comparison between them until time value is completely formed.
+    if (TimeUtil.validateTime(this.props.value, this.props.showSeconds) && TimeUtil.validateTime(prevStateTimeValue, this.props.showSeconds)) {
+      skipStateUpdate = this.hasTimeValueUpdated(prevState);
```

Comment:
If the time-input passes in a valid `value` prop, it would always get in here and prevents manually changing the time. You can go to this [example](https://terra-framework-deploye-pr-888.herokuapp.com/tests/terra-time-input/time-input/time-input/default-time) and attempt to manually change the time.

Reply:
this issue has been addressed with : 7d1a4a89def749924df1b0ac06c1ae5887a71939